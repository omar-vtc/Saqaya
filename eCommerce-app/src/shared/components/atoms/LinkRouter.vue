<script setup lang="ts">
import { defineProps } from "vue";

const props = defineProps<{
  className?: string;
  to: string;
  value: string;
  exact?: boolean;
}>();

// console.log(props.exact);
</script>

<template>
  <router-link
    :to="props.to"
    :class="props.className"
    v-bind="{ exact: props.exact }"
    >{{ props.value }}</router-link
  >
</template>
