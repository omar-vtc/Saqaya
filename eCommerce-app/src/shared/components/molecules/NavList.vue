<script setup lang="ts">
import { defineProps } from "vue";
import LinkRouter from "../atoms/LinkRouter.vue";
const props = defineProps<{
  divClassName?: string;
  className?: string;
  eleClassName?: string;
}>();
</script>
<template>
  <div :class="props.divClassName">
    <ul :class="props.className">
      <li :class="props.eleClassName">
        <LinkRouter
          :to="'/'"
          class-name="list-item__ele"
          value="Home"
          :exact="true"
        />
      </li>
      <li :class="props.eleClassName">
        <LinkRouter
          :to="'/products'"
          class-name="list-item__ele"
          value="Products"
          :exact="false"
        />
      </li>
      <li :class="props.eleClassName">
        <LinkRouter
          :to="'/contact-us'"
          class-name="list-item__ele"
          value="Contact Us"
          :exact="false"
        />
      </li>
    </ul>
  </div>
</template>
