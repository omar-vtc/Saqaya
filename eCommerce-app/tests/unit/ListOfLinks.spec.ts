import { mount } from "@vue/test-utils";
import NavList from "../../src/shared/components/molecules/NavList.vue";
import LinkRouter from "../../src/shared/components/atoms/LinkRouter.vue";
import { describe, expect, it } from "vitest";

describe("NavList.vue", () => {
  it("renders the correct number of LinkRouter items", () => {
    const wrapper = mount(NavList, {
      global: {
        stubs: {
          LinkRouter: {
            template: '<div class="stub">Link</div>',
          },
        },
      },
    });

    const items = wrapper.findAllComponents(LinkRouter);
    expect(items.length).toBe(3); // Home, Products, Contact Us
  });

  it("applies the correct classes from props", () => {
    const wrapper = mount(NavList, {
      props: {
        divClassName: "wrapper",
        className: "link-list",
        eleClassName: "link-item",
      },
      global: {
        stubs: {
          LinkRouter: true,
        },
      },
    });

    expect(wrapper.find("div").classes()).toContain("wrapper");
    expect(wrapper.find("ul").classes()).toContain("link-list");
    const items = wrapper.findAll("li");
    items.forEach((item) => {
      expect(item.classes()).toContain("link-item");
    });
  });

  it("matches snapshot", () => {
    const wrapper = mount(NavList, {
      global: {
        stubs: {
          LinkRouter: true,
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
