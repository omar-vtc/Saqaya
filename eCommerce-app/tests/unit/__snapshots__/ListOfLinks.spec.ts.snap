// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavList.vue > matches snapshot 1`] = `
"<div class="">
  <ul class="">
    <li class="">
      <link-router-stub to="/" class="list-item__ele" exact="true" value="Home"></link-router-stub>
    </li>
    <li class="">
      <link-router-stub to="/products" class="list-item__ele" exact="false" value="Products"></link-router-stub>
    </li>
    <li class="">
      <link-router-stub to="/contact-us" class="list-item__ele" exact="false" value="Contact Us"></link-router-stub>
    </li>
  </ul>
</div>"
`;
